go/grpcwebproxy/main.go

```diff
@@ -121,6 +127,24 @@ func main() {
 	// TODO(mwitkow): Add graceful shutdown.
 }
 
+func buildDebugServer(wrappedGrpc *grpcweb.WrappedGrpcServer, metricsHandler http.Handler) *http.Server {
+	return &http.Server{
+		WriteTimeout: *flagHttpMaxWriteTimeout,
+		ReadTimeout:  *flagHttpMaxReadTimeout,
+		Handler: http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+			if req.URL.Path == "/metrics" {
+				metricsHandler.ServeHTTP(resp, req)
+			} else if req.URL.Path == "/debug/requests" {
+				trace.Traces(resp, req)
+			} else if req.URL.Path == "/debug/events" {
+				trace.Events(resp, req)
+			} else {
+				wrappedGrpc.ServeHTTP(resp, req)
+			}
```

Comment:
This is great, but what do you think about making it a switch statement instead? It's a bit more Go-like:
```suggestion
			switch req.URL.Path {
			case "/metrics":
				metricsHandler.ServeHTTP(resp, req)
			case "/debug/requests":
				trace.Traces(resp, req)
			case "/debug/events":
				trace.Events(resp, req)
			default:
				wrappedGrpc.ServeHTTP(resp, req)
			}
```